File: apps/push-service/src/main/java/com/example/chat/push/infrastructure/sender/SlackPushSender.java
package com.example.chat.push.infrastructure.sender;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

import java.util.Map;

/**
 * Slack 푸시 발송자
 */
@Component
@RequiredArgsConstructor
@Slf4j
public class SlackPushSender implements PushSender {
    private final RestTemplate restTemplate;

    @Override
    public boolean support(String pushType) {
        return "SLACK".equalsIgnoreCase(pushType);
    }

    @Override
    public void send(String targetUserId, String title, String content) {
        String webhookUrl = "https://hooks.slack.com/services/...";
        Map<String, String> payload = Map.of("text", "*" + title + "*\n" + content);

        log.info("Sending Slack push to user {} via {}", targetUserId, webhookUrl);
        // restTemplate.postForEntity(webhookUrl, payload, String.class);
        log.info("Slack payload: {}", payload);
    }
}

File: apps/push-service/src/main/kotlin/com/example/chat/push/infrastructure/sender/SlackPushSender.kt
package com.example.chat.push.infrastructure.sender

import io.github.oshai.kotlinlogging.KotlinLogging
import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

private val logger = KotlinLogging.logger {}

@Component
class SlackPushSender(private val restTemplate: RestTemplate) : PushSender {
    override fun support(pushType: String): Boolean = pushType.uppercase() == "SLACK"

    override fun send(targetUserId: String, title: String, content: String) {
        // In a real system, you'd fetch the user's Slack Webhook URL from a database or service
        val webhookUrl = "https://hooks.slack.com/services/..."

        val payload = mapOf("text" to "*$title*\n$content")

        logger.info { "Sending Slack push to user $targetUserId via $webhookUrl" }

        // Simulating the call (commented out to avoid real 404/errors in background)
        // restTemplate.postForEntity(webhookUrl, payload, String::class.java)

        logger.info { "Slack payload: $payload" }
    }
}
